bin/program.py

```diff
@@ -99,7 +103,9 @@ def sanitizer():
 #
 # build() will return the (run_command, message) pair.
 class Program:
-    def __init__(self, problem: Problem, path: Path, deps=None, *, skip_double_build_warning=False):
+    def __init__(
+        self, problem: "Problem", path: Path, deps=None, *, skip_double_build_warning=False
```

Comment:
this `"Problem"` is quite ugly. Can't we extract the class definition to a near-empty file without further dependencies?

Reply:
Ok well; string types it is then.